src/lib/hooks/useAccountSwitcher.ts

```diff
@@ -23,16 +20,15 @@ export function useAccountSwitcher() {
       try {
         if (account.accessJwt) {
           closeAllActiveElements()
-          await selectAccount(account)
-
-          /*
-           * Mobile resets to Home on its own, so no need to handle mobile
-           * specifically.
-           */
           if (isWeb) {
-            navigation.navigate('Home')
+            // We're switching accounts, which remounts the entire app.
+            // On mobile, this gets us Home, but on the web we also need reset the URL.
+            // We can't change the URL via a navigate() call because the navigator
+            // itself is about to unmount, and it calls pushState() too late.
+            // So we change the URL ourselves. The navigator will pick it up on remount.
+            history.pushState(null, '', '/')
```

Comment:
I put this before `selectAccount` to emphasize that we're changing something for the action caused by `selectAccount` to "pick up" later. By itself this call doesn't re-render.